src/utils/message.js

```diff
@@ -1,5 +1,6 @@
 import { WORD_DECIMETER } from '../constants';
 
 export const messageToArrayOfWords = ( message ) => {
-  return message.trim().split( WORD_DECIMETER );
+  // the filter discards the empty strings that might get returned by split()
+  return message.trim().split( WORD_DECIMETER ).filter(Boolean);
```

Comment:
[this regular expression](https://github.com/ChatAnalyzer/chat-analyzer/pull/3/files#diff-a7748843fbda5167d065231e3a2e7a1fR1) is nice, but you can avoid having the filter (because you are looping twice, once for the split and once for the filter, which is costly if we have a lot of messages) if you negated it: 

```js
export const WORD_DECIMETER = /[^\b\s ()?!:,|\[\]+.-;"^'`´&-]+/g; //notice the global flag
```
and then do: 
```js
return message.match( WORD_DECIMETER );
```